apps/angular/src/app/features/anime/components/anime-table/anime-table.component.ts

```diff
@@ -0,0 +1,139 @@
+
+  /** Anime list. */
+  public readonly animeList$: Observable<readonly Anime[]>;
+
+  /** Displayed columns. */
+  public readonly displayedColumns = ['imageSrc', 'titleEnglish', 'titleJapanese', 'type', 'status', 'airedStart'] as const;
+
+  /** Number of records per page. */
+  public readonly pageSize$ = new BehaviorSubject(PAGE_SIZE_DEFAULT);
+
+  /** Anime Search field value. */
+  public searchValue = DEFAULT_SEARCH_VALUE;
+
+  /** Count of anime taken from the request. */
+  public readonly animeCount$ = new BehaviorSubject(DEFAULT_ANIME_COUNT);
+
+  /** Index of the current page.  */
+  public readonly pageIndex$ = new BehaviorSubject(INITIAL_PAGE);
+
+  /** Sorting anime on the page. */
+  public readonly pageSort$ = new BehaviorSubject(ORDERING_DEFAULT);
+
+  /** Filtering by type on the page. */
+  public readonly pageType$ = new BehaviorSubject([] as AnimeType[]);
+
+  /** Filtering by anime name. */
+  private readonly pageSearch$ = new BehaviorSubject(DEFAULT_SEARCH_VALUE);
+
+  /** Anime types. */
+  public readonly animeTypes = Object.values(AnimeType).filter(element => typeof element === 'string');
+
+  public constructor(
+    private readonly animeService: AnimeService,
+    private readonly searchParamsService: SearchParamsService,
+  ) {
+    this.animeList$ = combineLatest([
+      this.pageType$,
+      this.pageSearch$,
+      this.pageSort$,
+      this.pageIndex$,
+      this.pageSize$,
+    ]).pipe(
+      debounceTime(DEBOUNCE_TIME),
+      switchMap(([pageType, pageSearch, pageSort, pageIndex, pageSize]) => {
+        const params = this.searchParamsService.changeSearchParams(new PaginationParams({
+          page: pageIndex,
+          size: pageSize,
+          sort: pageSort,
+          search: pageSearch,
+          type: pageType,
+        }));
+        return this.animeService.getAnimeList(params).pipe(
+          tap(animeList => this.animeCount$.next(animeList.count)),
+          map(animeList => animeList.results),
+        );
+      }),
+    );
+  }
+
+  /**
+   * Changing the page numbering to the anime table.
+   * @param event Paginator event.
+   */
+  public onPaginateChange(event: PageEvent): void {
+    this.pageIndex$.next(event.pageIndex);
+    this.pageSize$.next(event.pageSize);
+  }
+
+  /**
+   * Anime filtering by type.
+   * @param selectTypeValue Select type value.
+   */
+  public onFilterType(selectTypeValue: MatSelectChange): void {
+    this.pageIndex$.next(INITIAL_PAGE);
+    this.pageType$.next(selectTypeValue.value);
+  }
+
+  /**
+   * Sort anime list.
+   * @param sort Selected sorting.
+   */
+  public onSortChange(sort: Sort): void {
+    if (!sort.active || sort.direction === 'asc') {
+      this.pageSort$.next(sort.active);
+    } else {
+      this.pageSort$.next(`-${sort.active}`);
+    }
+  }
+
+  /** Searching input form controller. */
+  public searchAnime(): void {
+    this.pageIndex$.next(INITIAL_PAGE);
+    this.pageSearch$.next(this.searchValue);
+  }
```

Comment:
Usually on Angular projects we use the `on` prefix to name event handlers
```suggestion
  /** Searching input form controller. */
  public onAnimeSearchChange(): void {
    this.pageIndex$.next(INITIAL_PAGE);
    this.pageSearch$.next(this.searchValue);
  }
```